pkg/prometheus/statefulset_test.go

```diff
@@ -1195,6 +1195,32 @@ func TestThanosObjectStorageFile(t *testing.T) {
 	}
 }
 
+func TestThanosBlockDuration(t *testing.T) {
+	testKey := "thanos-config-secret-test"
+
+	sset, err := makeStatefulSetFromPrometheus(monitoringv1.Prometheus{
+		Spec: monitoringv1.PrometheusSpec{
+			Thanos: &monitoringv1.ThanosSpec{
+				BlockDuration: "1h",
+				ObjectStorageConfig: &v1.SecretKeySelector{
+					Key: testKey,
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	found := false
+	for _, arg := range sset.Spec.Template.Spec.Containers[0].Args {
+		if arg == "--storage.tsdb.max-block-duration=1h" {
+			found = true
+		}
+	}
+	if !found {
+		t.Fatal("Thanos BlockDuration arg change not found")
+	}
```

Comment:
nit: print expected argument if the test fails
```suggestion
	found := false
	expectedBlockDuration := "--storage.tsdb.max-block-duration=1h"
	for _, arg := range sset.Spec.Template.Spec.Containers[0].Args {
		if arg == expectedBlockDuration {
			found = true
		}
	}
	if !found {
		t.Fatalf("expected to find Thanos BlockDuration argument %s in the Prometheus container", expectedBlockDuration)
	}
```

Reply:
Oops 😅 I was too late, I'll open a follow up PR tomorrow with the suggestion